client/src/components/Upload/UploadModalContent.vue

```diff
@@ -167,12 +168,50 @@ export default {
         currentFtp: function () {
             return this.currentUserId && this.ftpUploadSite;
         },
-
+        toData: function (items, history_id) {
+            var data = {
+                fetchRequest: null,
+                uploadRequest: null,
+            };
+            if (items && items.length > 0) {
+                var split = this.preprocess(items);
+                if (split.urls.length > 0) {
+                    data.fetchRequest = this.toFetchData(split.urls, history_id);
+                }
+                if (split.files.length > 0) {
+                    data.uploadRequest = this.toFileUploadData(split.files, history_id);
+                }
```

Comment:
This looks like both data.fetchRequest and data.uploadRequest could be set right? But then other places files would be ignored if fetchRequest is set.

Reply:
@jmchilton whoops! I wrote this when I was still thinking I would need to use this for Composites, but since Composites should fail if one piece of data is no good, I'll convert this to a simple if/else.